test/Microsoft.Framework.DesignTimeHost.Tests/PluginHandlerFacts.cs

```
@@ -0,0 +1,491 @@
+            error = Assert.Throws<InvalidOperationException>(
+                () => pluginHandler.ProcessMessage(unregisterPluginMessage, assemblyLoadContext));
+            Assert.Equal(expectedErrorMessage, error.Message, StringComparer.Ordinal);
+        }
+
+        [Fact]
+        public void ProcessMessage_PluginMessage_ProcessesMessages()
+        {
+            var assemblyLoadContext = CreateTestAssemblyLoadContext<MessageTestPlugin>();
+            var serviceProvider = new TestServiceProvider();
+            PluginMessageBroker.PluginMessageWrapperData messageBrokerData = null;
+            var pluginHandler = new PluginHandler(
+                serviceProvider, (data) => messageBrokerData = (PluginMessageBroker.PluginMessageWrapperData)data);
+            var registerPluginMessage = new PluginMessage
+            {
+                Data = new JObject
+                {
+                    { "AssemblyName", "CustomAssembly" },
+                    { "TypeName", typeof(MessageTestPlugin).FullName },
+                },
+                MessageName = "RegisterPlugin",
+                PluginId = RandomGuidId
+            };
+            var pluginMessage = new PluginMessage
+            {
+                Data = new JObject
+                {
+                    { "Data", "Hello Plugin" },
+                },
+                MessageName = "PluginMessage",
+                PluginId = RandomGuidId
+            };
+
+            pluginHandler.ProcessMessage(registerPluginMessage, assemblyLoadContext);
+            pluginHandler.ProcessMessage(pluginMessage, assemblyLoadContext);
+
+            Assert.NotNull(messageBrokerData);
+            Assert.Equal(RandomGuidId, messageBrokerData.PluginId);
+            var actualMessage = (string)messageBrokerData.Data;
+            Assert.Equal("Hello Plugin!", actualMessage, StringComparer.Ordinal);
+        }
+
+        [Fact]
+        public void ProcessMessage_PluginMessage_ThrowsWhenPluginNotRegistered()
```

Would rather be vocal about non-existent plugins so we aren't failing silently from VS' perspective. 

Plugin registration is still restrictive but more reliant on things like the `AssemblyLoadContext` to throw for invalid attempts.